### PR TITLE
BasicUI: check icons existence and replace with blank when needed

### PR DESCRIPTION
**Pull request by resetnow**
Signed-off-by: Vlad Ivanov <vlad@ivanov.email>

### CODE REVIEW

**Comment by resetnow** [comment]
This is the least sketchy implementation I could come up with, would gladly hear any suggestions.

File: extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/WebAppIconProvider.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * Helper class, provides icon names
+ * for sitemap widgets.
+ *
+ * @author Vlad Ivanov - Initial contribution
+ */
+public class WebAppIconProvider {
+    private static final String DEFAULT_ICON_SET = "classic";
+
+    private static List<IconProvider> iconProviders = new ArrayList<>();
+    private static ItemUIRegistry itemUIRegistry;
```
Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/WebAppIconProvider.java]: shouldn't be static, will fix it a bit later

**Comment by kaikreuzer** [comment]
Err, just to summarize: You have implemented your own icon provider, which acts as a proxy for the normal ones?

This looks really a bit nasty... Will have to think about it and come up with a better suggestion - but now it is too late at night already and I cannot think clearly anymore :-/

**Comment by resetnow** [comment]
Right — it's a proxy, so might as well name it so. Another option is to move this functionality into AbstractWidgetRenderer, but then a reference would need to be added to OSGi description file for every single renderer class.

**Comment by resetnow** [comment]
@kaikreuzer just checking — did you have a time to look for a better solution?

**Comment by kaikreuzer** [comment]
Not yet, sorry - will come back to you on this soon!

**Comment by kaikreuzer** [comment]
@resetnow Sorry for the long silence. Taking a step back, I would actually first like to understand how it comes that there are situations with missing icons. I suggest to do this discussion at https://github.com/eclipse/smarthome/issues/2994.

**Pull request closed without merging**